src/screens/HomeScreen/components/NameBarContainer.js

```diff
@@ -32,11 +31,20 @@ const NameBarContainer = () => {
   }, []);
 
   useEffect(() => {
-    getUserData(name)
-      .then((list) => list.wordList.map((word) => word.word))
-      .then((value) => setUserData(value));
+    getUserWordList(name).then((value) => setUserData(value));
```

Comment:
You could probably move `refreshList` to above this `useEffect` call and use it inside the effect.